scripts/run_dpo.py

```diff
@@ -105,13 +105,17 @@ def main():
     torch_dtype = (
         model_args.torch_dtype if model_args.torch_dtype in ["auto", None] else getattr(torch, model_args.torch_dtype)
     )
+    params_offloaded = False
+    if accelerator.state.deepspeed_plugin and accelerator.state.deepspeed_plugin.offload_param_device != "none":
+        params_offloaded = True
+
     model_kwargs = dict(
         revision=model_args.model_revision,
         trust_remote_code=model_args.trust_remote_code,
         use_flash_attention_2=model_args.use_flash_attention_2,
         torch_dtype=torch_dtype,
         use_cache=False if training_args.gradient_checkpointing else True,
-        device_map=get_kbit_device_map(),
+        device_map=get_kbit_device_map() if not params_offloaded else None,
```

Comment:
I just realised that since we only want the device map specified for QLoRA, we can define `quantization_config=get_quantization_config(model_args)` above `model_kwargs` and then do this which should cover all forms of ZeRO-3:

```suggestion
        device_map=get_kbit_device_map() if quantization_config is not None else None,
```